src/schemas.jl

```diff
@@ -259,21 +261,30 @@ schema violations, see [`Legolas.find_violations`](@ref).
 
 See also: [`Legolas.find_violations`](@ref), [`Legolas.validate`](@ref), [`Legolas.complies_with`](@ref)
 """
-find_violation(::Tables.Schema, sv::SchemaVersion) = throw(UnknownSchemaVersionError(sv))
+function find_violation(ts::Tables.Schema, sv::SchemaVersion)
+    Base.depwarn(
+        "`find_violation(ts, sv)` is deprecated, use: `let v = find_violations(ts, sv); " *
+        "isempty(v) ? nothing : v; end` instead.",
+        :find_violation,
+    )
```

Comment:
Could use `Base.@deprecate` here but the expression it prints ends up expanding the ternary expression which makes this more verbose than I intended